src/app/file/bmp_format.cpp

```diff
@@ -151,26 +243,74 @@ static int read_bmfileheader(FILE *f, BITMAPFILEHEADER *fileheader)
 /* read_win_bminfoheader:
  *  Reads information from a BMP file header.
  */
-static int read_win_bminfoheader(FILE *f, BITMAPINFOHEADER *infoheader)
+static int read_win_bminfoheader(FILE *f, BITMAPINFOHEADER *infoheader, int biSize)
 {
   WINBMPINFOHEADER win_infoheader;
 
+  if (biSize != 16 &&
+      biSize != 24 &&
+      biSize != WININFOHEADERSIZE &&
+      biSize != BV2INFOHEADERSIZE &&
+      biSize != BV3INFOHEADERSIZE &&
+      biSize != 60 &&
+      biSize != 64) return -1;
```

Comment:
I would suggest to define some extra macros for those sizes (16, 24, 60, 64). Also is there a place/website where we can find some information about these header sizes?
```suggestion
  if (biSize != 16 &&
      biSize != 24 &&
      biSize != WININFOHEADERSIZE &&
      biSize != BV2INFOHEADERSIZE &&
      biSize != BV3INFOHEADERSIZE &&
      biSize != 60 &&
      biSize != 64)
    return -1;
```
:point_up: make the return more visible moving it to its own line.